test-poc/framework/src/main/java/org/keycloak/test/framework/server/EmbeddedKeycloakTestServer.java

```diff
@@ -11,20 +13,34 @@ public class EmbeddedKeycloakTestServer implements KeycloakTestServer {
 
     private Keycloak keycloak;
 
+    private TestDatabase database;
+
     @Override
-    public void start(KeycloakTestServerConfig serverConfig) {
+    public void start(KeycloakTestServerConfig serverConfig, TestDatabase testDatabase) {
+        database = testDatabase;
         serverConfig.adminUserName().ifPresent(username -> System.setProperty("keycloakAdmin", username));
         serverConfig.adminUserPassword().ifPresent(password -> System.setProperty("keycloakAdminPassword", password));
 
         List<String> rawOptions = new LinkedList<>();
         rawOptions.add("start-dev");
-//        rawOptions.add("--db=dev-mem"); // TODO With dev-mem there's an issue as the H2 DB isn't stopped when restarting embedded server
         rawOptions.add("--cache=local");
 
         if (!serverConfig.features().isEmpty()) {
             rawOptions.add("--features=" + String.join(",", serverConfig.features()));
         }
 
+        DatabaseConfig databaseConfig = database.getDatabaseConfig();
+
+        //rawOptions.add("--db=" + databaseConfig.vendor()); // TODO With dev-mem there's an issue as the H2 DB isn't stopped when restarting embedded server
+        if (databaseConfig.isExternal()) {
+            rawOptions.add("--db-url-host=" + databaseConfig.urlHost());
+            rawOptions.add("--db-username=" + databaseConfig.username());
+            rawOptions.add("--db-password=" + databaseConfig.password());
+        } else {
+            testDatabase.start();
```

Comment:
KeycloakTestServer shouldn't be starting the database, that is the database suppliers job